include/clang/Lex/MacroInfo.h

```diff
@@ -515,6 +510,9 @@ class ModuleMacro : public llvm::FoldingSetNode {
     ID.AddPointer(II);
   }
 
+  /// Get the name of the macro.
+  IdentifierInfo *getName() const { return II; }
```

Comment:
Yes, it seems fine to expose this; the name is part of the module macro abstraction and not just an implementation detail. I'm a little surprised we don't provide such an accessor already :)